optimum/intel/ipex/modeling_base.py

```diff
@@ -86,10 +87,37 @@ def _is_patched_with_ipex(model, task):
 
 
 def _prepare_inputs_for_ipex_model(model, task, use_cache):
-    if task in _IPEX_EXPORTED_GENERATION_TASKS and _is_patched_with_ipex(model, task):
-        return get_dummy_input(model, return_dict=True)
-    else:
-        return prepare_jit_inputs(model, task, use_cache)
+    sample_inputs = prepare_jit_inputs(model, task, use_cache)
+    if (
+        task in _IPEX_EXPORTED_GENERATION_TASKS
+        and _is_patched_with_ipex(model, task)
+        and "past_key_values" in sample_inputs
+    ):
+        # Only consider llama for now
+        assert len(sample_inputs["past_key_values"][0][0].shape) == 4
+        max_position = model.config.max_position_embeddings
+        batch_size = sample_inputs["input_ids"].shape[0]
+        past_length = sample_inputs["past_key_values"][0][0].shape[2]
+        num_attn = sample_inputs["past_key_values"][0][0].shape[1]
+        d_k = sample_inputs["past_key_values"][0][0].shape[-1]
+        dtype = sample_inputs["past_key_values"][0][0].dtype
+
+        num_layers = len(sample_inputs["past_key_values"])
+        beam_idx_tmp = torch.zeros((max_position, batch_size), dtype=torch.long).contiguous()
+        past_key_values = tuple(
+            [
+                (
+                    torch.zeros(1, past_length, past_length, 1, dtype=torch.long).contiguous(),
+                    torch.zeros(max_position, batch_size, num_attn, d_k, dtype=dtype).contiguous(),
+                    torch.zeros(max_position, batch_size, num_attn, d_k, dtype=dtype).contiguous(),
+                    beam_idx_tmp,
+                )
+                for i in range(num_layers)
+            ]
+        )
```

Comment:
why not have our own `LlamaDummyPastKeyValuesGenerator` https://github.com/huggingface/optimum/blob/v1.21.2/optimum/utils/input_generators.py#L1122 ?

Reply:
Done.